src/System.IO.Ports/src/System/IO/Ports/SerialPort.Linux.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.ComponentModel;
+using System.Collections.Generic;
+using System.IO;
+
+namespace System.IO.Ports
+{
+    public partial class SerialPort : Component
+    {
+        public static string[] GetPortNames()
+        {
+            const string sysDir = "/sys/class/tty";
+
+            if (Directory.Exists(sysDir))
+            {
+                // /sys is mounted. Let's explore tty class and pick active nodes.
+                List<string> ports = new List<string>();
+                DirectoryInfo di = new DirectoryInfo(@"/sys/class/tty");
+                var entries = di.EnumerateFileSystemInfos(@"*", SearchOption.TopDirectoryOnly);
```

Comment:
I think that `Directory.GetFiles` will be better: you can avoid the allocations of FileInfo objects, set the capacity of ports at creation and use Path.GetFileName to get the name for the comparison below only when the one that uses the full path is not true. 

@JeremyKuhne any better option?

Reply:
GetFiles does not work as this is full os symbolic links.